pageserver/src/consumption_metrics/metrics.rs

```diff
@@ -477,6 +499,15 @@ impl TimelineSnapshot {
             });
         }
 
+        // Compute the PITR history size.
+        //
+        // TODO: verify that the GC cutoffs don't severely regress, e.g. to 0 such that we bill the
+        // entire history. Also verify that it's okay for this to regress on restart, unlike e.g.
+        // written_size above.
+        let pitr_history_size_key = MetricsKey::pitr_history_size(tenant_id, timeline_id);
+        let pitr_history_size = self.last_record_lsn.saturating_sub(self.gc_cutoffs.time);
```

Comment:
Thinking about edge case of someone with pitr_interval=0 -- we'll end up charging them some nonzero amount as `gc_cutoffs.time` as last_record_lsn would advance a bit since we updated `gc_cutoffs.time`.

Maybe worth special casing this to always return 0 for pitr_history_size if pitr_interval is zero?

Reply:
Good catch, done.